components/ai_chat/resources/page/components/edit_input/index.tsx

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import * as React from 'react'
+
+import Button from '@brave/leo/react/button'
+import { getLocale } from '$web-common/locale'
+
+import styles from './style.module.scss'
+
+interface Props {
+  text: string
+  onSubmit: (text: string) => void
+  onCancel: () => void
+  isSubmitDisabled: boolean
+}
+
+function EditInput(props: Props) {
+  const [text, setText] = React.useState(props.text)
+  const handleKeyDown = (e: React.KeyboardEvent<HTMLTextAreaElement>) => {
+    if (e.key === 'Enter' && !e.shiftKey && !e.nativeEvent.isComposing) {
```

Comment:
I think this handler needs to check `props.isSubmitDisabled`

Reply:
Added in the final squashed commit.